include/firmware/common/spi_comms.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include <span>
+
+#include "stm32g4xx_hal_conf.h"
+
+namespace spi {
+class Spi {
+  public:
+    static constexpr auto BUFFER_SIZE = 5;
+    using BufferType = std::array<uint8_t, BUFFER_SIZE>;
+    Spi();
+    void send_command(const BufferType& aTxBuffer, uint32_t& data,
```

Comment:
I think it may be a better idea to not have this interface differentiate between `data` and `status` and just take and provide buffers of bytes. We want the things that put the bytes in buffers to be in the testable layers anyway, and regardless of where the serialization/deserialization happens these functions aren't actually parsing or building the spi messages themselves (i.e., this class doesn't know what the various registers or their content formats are) so I think we should just go all the way.